assets/js/utils/rtcUtils.ts

```diff
@@ -294,22 +307,50 @@ const filterDevices = (allDevices: MediaDeviceInfo[], type: string) => {
   return allDevices.filter((device) => device.deviceId != "default" && device.kind == type);
 };
 
-const addOrReplaceTrack = (
+export const addOrReplaceTrack = (
   client: Client,
   track: MediaStreamTrack,
-  playerCallback: (sourceType: SourceType) => void
+  setPresenterStreams: React.Dispatch<React.SetStateAction<PresenterStreams>>
 ) => {
-  if (!presenterStreams[client.email]) presenterStreams[client.email] = new MediaStream();
-  const curTrack = findTrackByType(client, track.kind as SourceType);
+  setPresenterStreams((prev) => {
+    const presenterStreams = { ...prev };
+    if (!presenterStreams[client.email])
+      presenterStreams[client.email] = {
+        name: client.name,
+        email: client.email,
+        stream: new MediaStream(),
+      };
 
-  if (curTrack && curTrack.id !== track.id && curTrack.id !== mergedScreenRef.screenTrack?.id) {
-    curTrack.stop();
-  }
+    const curTrack = findTrackByType(client, track.kind as SourceType, presenterStreams);
+
+    if (curTrack && curTrack.id !== track.id && curTrack.id !== mergedScreenRef.screenTrack?.id) {
+      curTrack.stop();
+    }
+
+    curTrack && presenterStreams[client.email].stream.removeTrack(curTrack);
+    presenterStreams[client.email].stream.addTrack(track);
+
+    return presenterStreams;
+  });
+};
+
+const removeTrack = (
+  client: Client,
+  track: MediaStreamTrack,
+  setPresenterStreams: React.Dispatch<React.SetStateAction<PresenterStreams>>
+) => {
+  setPresenterStreams((prev) => {
+    const presenterStreams = { ...prev };
+    const curTrack = findTrackByType(client, track.kind as SourceType, presenterStreams);
 
-  curTrack && presenterStreams[client.email].removeTrack(curTrack);
-  presenterStreams[client.email].addTrack(track);
```

Comment:
Are you sure about this removed line?

Reply:
Yep, we don't want to add the track in the `removeTrack` function.